topics/atomic-slot-migration.md

```diff
@@ -0,0 +1,258 @@
+       |            ...             |                                                |                                 |
+       |                            |                                            (performs takeover &                  |
+       |-CLUSTER GETSLOTMIGRATIONS->|                                             propagates topology)                 |
+       |<-----{"state": "..."}------|                                                |                                 |
+       |            ...             |                                                |------- SYNCSLOTS FINISH ------->|
+       |                      (finds out about topology                              |                                 |
+       |                       change & marks migration done)                        |                                 |
+       |                            |                                                |                                 |
+       |-CLUSTER GETSLOTMIGRATIONS->|                                                |                                 |
+       |<---{"state": "success"}----|                                                |                                 |
+```
+
+Since the `CLUSTER SYNCSLOTS` command is considered an internal command, the
+subcommands and semantics are subject to change in future versions.
+
+## Benefits of Atomic Slot Migration
+
+By replicating the slot contents and atomically transferring ownership, Atomic
+Slot Migration provides many desirable properties over the previous mechanism:
+
+1. **Clients are unaware**: Since the entire hash slot is replicated before any
+   cleanup is done on the source node, clients are completely unaware of the
+   slot migration, and no longer need to follow `ASK` redirections and handle
+   `TRYAGAIN` errors for multi-key operations.
+2. **Keys no longer need to be atomically moved**: Collections are moved in
+   chunks, preventing the reliability problems previously encountered when
+   dumping and restoring a large collection.
+3. **A migration can easily be rolled back on cancellation or failure**: Since
+   the hash slots are placed in a staging area, we can easily wipe them
+   independently of the rest of the database. Since this state is not
+   broadcasted to the cluster, ending the migration is as simple as cleaning up
+   the staging area and marking the migration as cancelled. Many failures, like
+   out-of-memory, failover, or network partition can be handled completely by
+   the engine, and only require the operator to check the result through
+   `CLUSTER GETSLOTMIGRATIONS`.
+4. **Greatly improved slot migration latency**: Valkey is already
+   highly-optimized for replication. By batching the slot migrations and
+   replicating their contents, the end-to-end migration latency can improve by
+   as much as 31x when compared to using `CLUSTER SETSLOT` through the
+   valkey-cli.
```

Comment:
We probably want to update https://valkey.io/topics/cluster-spec/ to better highlight why atomic slot migration is better.